File: test-suite-kotlin/src/test/kotlin/io/micronaut/el/test/Book.kt
package io.micronaut.el.test

import io.micronaut.context.annotation.Executable
import io.micronaut.core.annotation.Introspected

@Introspected
class Book(var title: String, val category: String, var unitPrice: Double) {

    val tags: List<String>
        get() = listOf("new", "sale", "b")

    @Executable
    fun discounted(percent: Double): Double = unitPrice - (unitPrice * percent / 100)

    @Executable
    fun count(predicate: java.util.function.Predicate<String>): Long = tags.stream().filter(predicate).count()
}

File: doc-examples/example-python/src/test/python/example/Book.py
from dataclasses import dataclass

from micronaut.context.annotation import Executable
from micronaut.core.annotation import Introspected


@Introspected  # <1>
@dataclass(frozen=True)
class Book:
    title: str  # <2>
    category: str
    unitPrice: float

    @Executable  # <3>
    def discounted(self, percent: float) -> float:
        return self.unitPrice * (100 - percent) / 100
